src/main/haskell/kore/src/Kore/AST/Common.hs

```diff
@@ -964,7 +966,7 @@ data Pattern level variable child where
     CeilPattern
         :: !(Ceil level child) -> Pattern level variable child
     DomainValuePattern
-        :: !(DomainValue Object child) -> Pattern Object variable child
+        :: !(DomainValue Object (Fix (Pattern Meta Variable))) -> Pattern Object variable child
```

Comment:
With this constructor, I don't think `Pattern` is going to do what we want. Traversals (`cata` or even just `project`) are not going to be able to descend into this child tree. The caller will have to write a second traversal inside the first just to deal with this constructor, defeating the purpose of using `Pattern` as a base functor.

Reply:
@ttuegel I know this patttern will not be able to be descended automatically, but I think it is acceptable to pay this price to be able to have a proper type for the child of a domain value.

The problem I was trying to solve here is that having it be a child like for any other pattern would make it impossible to represent object-only patterns containing domain values, because the child of a domainvalue must be a meta-pattern of sort  `#String`.

If you can think of an alternative solution, I would be glad to hear it.

Reply:
Also, I think that we do not usually want to descend into a domain value's child, except in a few narrow cases (e.g. serialization), or after it was lifted to meta, when it's not a DomainValuePattern anymore.